sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4531,15 +4532,10 @@ enqueueMemImageCommandHelper(pi_command_type CommandType, pi_queue Queue,
 
   ze_event_handle_t ZeEvent = nullptr;
   if (Event) {
-    auto Res = piEventCreate(Queue->Context, Event);
+    auto Res = createEventAndAssociateQueue(Queue, Event, CommandType,
+                                            ZeCommandList, &ZeEvent, false);
```

Comment:
We DO have to lock the queue earlier to operate on the queue in line 4529. I think it is reasonable to have it locked for longer (until here for sure), and throughout the entire interface to avoid this locked/unlocked thing altogether.

Reply:
Actually, the helper function takes a boolean parameter "QueueAcquired".
In this particular case, setting it false means the helper function will call piQueueRetain which acquires lock for a short period of time and release it immediately.
So, we don't need to move the lock_guard above.

Reply:
But I am saying it is not correct not to have the lock acquired at line 4529 above your code. So your code would just use that lock, and thus remove the QueueAcquired completely.

Reply:
Actually, the comment of at line 4515 says the caller must have hold a lock on the Queue already.
So, we must not try to hold a lock again.

Reply:
That's exactly what I say. I expect that each caller of the "createEventAndAssociateQueue" should already have Queue locked. This is true in most cases already, and the rest of the cases can be restructured so that this is true.

Reply:
Sorry, I get it now. I will change the code to expect the caller to have already acquired lock.
"false" was a bug, I already fixed it and updated the code.

Reply:
Done. Thanks.